simulator/modules/core/perf_sim.cpp

```diff
@@ -112,7 +112,8 @@ void PerfSim<ISA>::dump_statistics() const
     auto frequency = double{ curr_cycle} / time; // cycles per millisecond = kHz
     auto ipc = 1.0 * executed_instrs / double{ curr_cycle};
     auto simips = executed_instrs / time;
-    auto mispredict_rate = 1.0 * branch.get_mispredict_rate();
+    auto decode_mispredict_rate = 1.0 * get_mispredict_rate( decode.get_jumps_num(), decode.get_mispredictions_num());
+    auto branch_mispredict_rate = 1.0 * get_mispredict_rate( decode.get_jumps_num(), branch.get_mispredictions_num());
```

Comment:
Decode and Branch stages do not execute the same amount of branches — there are pipeline flushes. It may lead to some mysteries like rate >100%. Usually to make meaningful statistic, you have to put both of your probe wires to the same place.